pkg/network/dns.go

```diff
@@ -116,7 +114,7 @@ func (dr *DnsResolver) DnsResolve(dnsAddr string, dnsLookupFamily v2.DnsLookupFa
 		//try from first server by default
 		for _, addr := range addrs {
 			msg.SetQuestion(addr, dnsQueryType)
-			msg.SetEdns0(4096, true)
+			msg.SetEdns0(defaultUDPBufferSize, true)
```

Comment:
扩大这个值不需要进行判断吗？

Reply:
> 扩大这个值不需要进行判断吗？

这里是指默认还是不变，动态扩容么？这地方没办法等收到response之后再进行设置，这里设置的是个默认大小，如果response小于这个值是没关系的

Reply:
了解

Reply:
这里应该是给查询 query 带上这个扩展描述，不太清楚是否解决了原始的问题呢？

> When DNS response is larger than 512, strict dns resolve result will be truncated.

这里是说那一侧会 truncated 呢？意思是，如果查询 query 没有这个 option，dns server 会 truncated 么？有点迷惑呢~

Reply:
这里是修改接收端buffer大小，查询不会修改，是接收到的dns response可能会被截断，比如response消息最大4096，实际大小1024，接收buffer当前是默认的512，只能解析出部分的结果

Reply:
哦，明白了，是我搞错了，这给确实不是 query 的扩展描述，只是这个 SetEdns0 API 名和注解看着比较费解。

我看了下，或许可以直接改 client 的 [UDPSize](https://github.com/mosn/mosn/blob/bb6845e6d8a34c0354b84d3f8f53edd94ddbefcc/pkg/network/dns.go#L85) 呢，效果应该是一样的~
看起来就直观一些，当然现在这个也正确，只是 API 看起来有些费解~

Reply:
这里我之前的理解有误，SetEdns0的注释：
```
// SetEdns0 appends a EDNS0 OPT RR to the message.
// TSIG should always the last RR in a message.
func (dns *Msg) SetEdns0(udpsize uint16, do bool) *Msg {
```
实际是会给msg加上选项，server端如果收到的msg没有特殊选项消息会从server端截断，
```
	// If EDNS0 is used use that for size.
	if opt != nil && opt.UDPSize() >= MinMsgSize {
		co.UDPSize = opt.UDPSize()
	}
	// Otherwise use the client's configured UDP size.
	if opt == nil && c.UDPSize >= MinMsgSize {
		co.UDPSize = c.UDPSize
	}
```
同时，如果有opt，则client的buffer也会相应调整，否则才使用client的buffer size

Reply:
那看起来这个 SetEdns0 就有两个用处了：
1. 当做一个 rr option 发给 server
2. 也用作客户端收到 response 后的 size 限制

这样的话，还是用 SetEdns0 更好一些了。可以考虑在代码里加上一些注释，这个 API 确实比较费解

Reply:
done